com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js

```diff
@@ -753,6 +777,12 @@ var Module = {
 
     setStatus: function(text) { console.log(text); },
 
+    isDMFSSupported: (function() {
```

Comment:
Maybe a silly question but what does DMFS stand for?

Reply:
I just saw the prefix DM all over, happy to call it whatever. :)

Reply:
I think dmfs is fine since we will have to support this.
Since this is now an "api", we should also document it as well.
I.e. what functions are supported/required.
Is there like a base class we could model it on?

Reply:
I'm struggling a bit with mapping in my mind how this will work and why the approach with dmfs is slightly different from the current one. Can we refactor it somehow so there are fewer branches? If I were to change anything in the functions of GameArchiveLoader I'd be worried that I would break things for one or the other solution. We also don't have any tests for any of this code which makes it even harder to reason about.

Reply:
The reason the code had to be refactored is because we need to load the _wasm.js first as that is where the FS comes from and I need FS to figure out if we already had it on disk.

As for supporting any API at defold I am not thinking there is any new API, FS already supports mounts and this is just making use of those from emscripten. The only real difference is loading the _wasm.js first and mounting this instead of IDBFS.

Reply:
> The reason the code had to be refactored is because we need to load the _wasm.js first as that is where the FS comes from and I need FS to figure out if we already had it on disk

Thanks, that explains a lot!

> Agree testing coverage would be good, though unsure how that would be done given we cannot have file system access in general browser case. I'm happy to help out with a test, but just not sure where it would go to get the contents or how to structure the test

Yeah, I don't know where to start with this either. Let's hold off on it for now.

> but it seemed rather intention to do that in parallel.

Yes, we want to reduce loading time as much as possible so this can't be changed in the default case.